proposals/adrianhopebailie-201908.md

```diff
@@ -0,0 +1,639 @@
+| toCurrency   | 1           | `Currency`  | Currency to which the conversion is being madre  |
+
+#### `RateValue` data type:
+
+The API data type `RateValue` is a JSON String consisting of digits only and an
+optional period. Negative numbers and leading zeroes are not allowed.
+
+##### `RateValue` Regular Expression
+
+The regular expression for restricting an Rate is
+`^(([1-9][0-9]{0,5}))([.][0-9]{0,8}[1-9])?$`
+
+#### Add `RequiredDataList`, `ProvidedDataList` and `ProvidedData`
+
+The `RequiredDataList` data model is defined as:
+
+| Data Element | Cardinality | Type             | Description                                                                    |
+| ------------ | ----------- | ---------------- | ------------------------------------------------------------------------------ |
+| requiredData | 1..256      | `String(1..128)` | Names of data elements required by the participant to complete the transaction |
+
+The `ProvidedDataList` data model is defined as:
+
+| Data Element | Cardinality | Type           | Description                                                                    |
+| ------------ | ----------- | -------------- | ------------------------------------------------------------------------------ |
+| providedData | 1..256      | `ProvidedData` | Names of data elements required by the participant to complete the transaction |
+
+The `ProvidedData` data model is defined as:
+
+| Data Element | Cardinality | Type              | Description                                                                   |
+| ------------ | ----------- | ----------------- | ----------------------------------------------------------------------------- |
+| dataKey      | 1           | `String(1..128)`  | Names of data element provided by the participant to complete the transaction |
+| dataValue    | 1           | `String(1..2056)` | Value of data element required by the participant to complete the transaction |
+
+#### Fees and Rates
+
+With each successive `POST /quotes` request in a multi-hop transaction the
+participants list will be added to by the participant processing the request.
+
+The participant MUST specify:
+
+1. the currency of the transfer it will make to the next participant
+2. the fee it will charge
+3. the rate of exchange it will apply
+
+If the `fee` is denominated in the same currency as the `transferCurrency` then
+the `fee` will be applied after applying the conversion at the specified `rate`.
+If the currency of the `fee` is not the same as the `transferCurrency` then it
+MUST be the same as the transfer that will be received by this participant (i.e.
+the `transferCurrency` of the preceding participant in the list). In this case,
+the fee is applied to the incoming transfer amount before the conversion.
+
+#### Regulatory Data
+
+When preparing a quote, each participant will indicate the data that it requires
+by putting the names data fields into the `dataRequired` list. It MUST also
```

Comment:
minor semantic issue: should this be `requiredData` instead? I know there is a `RequiredData` model, so I found this field name a little confusing.